### PR TITLE
Support S3 Account Settings

### PR DESCRIPTION
**Pull request by bbernays**
fixes: #282

### CODE REVIEW

File: resources/s3_accounts.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	var accountConfig s3control.GetPublicAccessBlockInput
+	accountConfig.AccountId = aws.String(c.AccountID)
+	s3AccountConfig.ConfigExists = false
+	resp, err := svc.GetPublicAccessBlock(ctx, &accountConfig, func(options *s3control.Options) {
+		options.Region = c.Region
+	})
+	if err != nil {
+		// If we received any error other than NoSuchPublicAccessBlockConfiguration, we return and error
+		var ae smithy.APIError
+		if errors.As(err, &ae) && ae.ErrorCode() != "NoSuchPublicAccessBlockConfiguration" {
```
Comment (yevgenypats) [inline resources/s3_accounts.go:79]: what will happen if it is returning `NoSuchPublicAccessBlockConfiguration` will resp be `nil`?

**Review by yevgenypats — DISMISSED** [review]
Looks great, I love the `ConfigExist` attribute, this will make queries much more simpler. 
Only mock and e2e tests are missing.

File: resources/s3_accounts.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+		s3AccountConfig.BlockPublicAcls = resp.PublicAccessBlockConfiguration.BlockPublicAcls
+		s3AccountConfig.BlockPublicPolicy = resp.PublicAccessBlockConfiguration.BlockPublicPolicy
+		s3AccountConfig.IgnorePublicAcls = resp.PublicAccessBlockConfiguration.IgnorePublicAcls
+		s3AccountConfig.RestrictPublicBuckets = resp.PublicAccessBlockConfiguration.RestrictPublicBuckets
+	} else {
+		s3AccountConfig.BlockPublicAcls = false
+		s3AccountConfig.BlockPublicPolicy = false
+		s3AccountConfig.IgnorePublicAcls = false
+		s3AccountConfig.RestrictPublicBuckets = false
+	}
```
Comment (roneli) [inline resources/s3_accounts.go:93]:
```suggestion
	if resp != nil {
		  res <- s3AccountConfig{resp.PublicAccessBlockConfiguration, true}
		}
```

File: resources/s3_accounts.go (hunk 1 above)
Comment (roneli) [reply resources/s3_accounts.go:79]: If err == nil, resp should be != nil

**Review by roneli — CHANGES_REQUESTED** [review]
LGTM, had one suggestion + as @yevgenypats we should add e2e + mock tests

File: resources/s3_accounts.go (hunk 1 above)
Comment (disq) [reply resources/s3_accounts.go:79]: Use the const `s3control.ErrCodeNoSuchPublicAccessBlockConfiguration` instead of string?

Comment (bbernays) [reply resources/s3_accounts.go:79]: @disq - That seems to have been removed from the s3control package

Comment (bbernays) [reply resources/s3_accounts.go]: @disc- I have updated the error handling to use the native types from within the s3control/types package

File: resources/s3_accounts.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		// If we received any error other than NoSuchPublicAccessBlockConfiguration, we return and error
+		var nspabc *s3controlTypes.NoSuchPublicAccessBlockConfiguration
+		if !errors.As(err, &nspabc) {
+			return err
+		}
+		res <- s3AccountConfig
+	} else {
+		res <- S3AccountConfig{*resp.PublicAccessBlockConfiguration, true}
+	}
+
```
Comment (roneli) [inline resources/s3_accounts.go:86]:
```suggestion
		if errors.As(err, &nspabc) {
		  res <- s3AccountConfig{nil, false}
		}
	}
	
	res <- S3AccountConfig{*resp.PublicAccessBlockConfiguration, true}

```

**Review by roneli — APPROVED** [review]
LGTM, added one suggestion to make the flow cleaner, I think you can also add account id to the initialization and remove initialization in lines 69-71.

Still missing mock tests for sanity + e2e tests I think no terraform is required, just a fetching the data and verifying values worked.

**Review by disq — APPROVED** [review]

**Pull request merged**